docs/development/releases-and-deployment.md

```diff
@@ -0,0 +1,102 @@
+# Spark release / deployment guidelines
+
+## GitHub releases
+
+* https://github.com/Midburn/Spark/releases
+* There should always be 1 draft release against master branch
+* This draft release is updated on each merge or commit to master.
+
+### Publishing a release
+
+* When release is ready (e.g. have enough content / is stable enough)- you can publish it
+* **Before publishing on GitHub** - you should update the [package.json](/package.json) "version" attribute to match the GitHub release you are about to publish
+* merge the change in package.json to master
+* don't forget to push!
+* now, you can publish the release in GitHub
+
+## Manual Deployment
+
+### Create the deployment package
+
+* ```npm run build --file=<OUTPUT_FILE_NAME>```
+  * We just tar.gz the source directory while excluding some files.
+  * might add some more stuff in the future - basically we want as much work done here as possible so deployment is quick
+
+### Upload the deployment package to slack
+
+We use slack as an easy to use repository of deployment packages.
+
+* ```npm run upload --file=<PACKAGE_FILE_NAME> --token=<SLACK_API_TOKEN>```
+  * creates a file called <PACKAGE_FILE_NAME>.slack_url containing the private package url
+
+### Download a package from the private package url
+
+* ```npm run download --file=<OUTPUT_FILE_NAME> --url=<PACKAGE_PRIVATE_URL> --token=<SLACK_API_TOKEN>```
```

Comment:
Perhaps it's worth it to group the relevant npm actions with a deploy: prefix? (`npm run deploy:upload`, `npm run deploy:build`, ...?) Not sure if makes sense, just something to think about.